pandaserver/dataservice/finisher.py

```diff
@@ -0,0 +1,304 @@
+                data=xml_data,
+            )
+
+    def check_file_status(self, job):
+        """
+        This function checks the status of the files for the job.
+
+        Parameters:
+        job (JobSpec): The job specification.
+
+        Returns:
+        tuple: A tuple containing three elements:
+            - bool: True if all files are ready, False otherwise.
+            - list: A list of failed files.
+            - list: A list of files with no output.
+        """
+        tmp_log = LogWrapper(_logger,
+                             f"check_file_status-{datetime.datetime.now(datetime.timezone.utc).replace(tzinfo=None).isoformat('/')}")
+        failed_files = []
+        no_out_files = []
+        for file in job.Files:
+            if file.type in ('output', 'log'):
+                if file.status == "failed":
+                    failed_files.append(file.lfn)
+                elif file.status == "nooutput":
+                    no_out_files.append(file.lfn)
+                elif file.status != "ready":
+                    tmp_log.debug(f"Job: {job.PandaID} file:{file.lfn} {file.status} != ready")
+                    return False, failed_files, no_out_files
+        return True, failed_files, no_out_files
+
+    # main
+    def run(self):
+        """
+        Starts the thread to finish transferring jobs
+        """
+        tmp_log = LogWrapper(_logger,
+                             f"run-{datetime.datetime.now(datetime.timezone.utc).replace(tzinfo=None).isoformat('/')}")
+        # start
+        try:
+            by_call_back = False
+            if self.job is None:
+                by_call_back = True
+                tmp_log.debug(f"start: {self.dataset.name}")
+                tmp_log.debug(f"callback from {self.site}")
+                # FIXME when callback from BNLPANDA disappeared
+                if self.site == "BNLPANDA":
+                    self.site = "BNL-OSG2_ATLASMCDISK"
+                # instantiate site mapper
+                site_mapper = SiteMapper(self.task_buffer)
+                # get computing_site/destination_se
+                computing_site, destination_se = self.task_buffer.getDestSE(self.dataset.name)
+                if destination_se is None:
+                    # try to get computing_site/destination_se from ARCH to delete sub
+                    # even if no active jobs left
+                    computing_site, destination_se = self.task_buffer.getDestSE(self.dataset.name, True)
+                    if destination_se is None:
+                        tmp_log.error(f"cannot get source/destination for {self.dataset.name}")
+                        tmp_log.debug(f"end: {self.dataset.name}")
+                        return
+                tmp_log.debug(f"src: {computing_site}")
+                tmp_log.debug(f"dst: {destination_se}")
+                # get corresponding token
+                tmp_source_site_spec = site_mapper.getSite(computing_site)
```

Comment:
I'm getting a bit confused about the usage of site, computing_site, se. How does it work that you can call getSite with a computing_site (PQ?) or with a se (RSE?). Would be good to review this and if this is how it works, then add some explanation

Reply:
I tried to keep the original logic as it was as I wasn't sure if something is obsolete or no...